### PR TITLE
hammer: osdc/Objecter: If osd full, it should pause read op which w/ rwordered flag

### PR DESCRIPTION
**Pull request by shinobu-x**
http://tracker.ceph.com/issues/19141

### CODE REVIEW

**Comment by shinobu-x** [comment]
retest this please

**Comment by shinobu-x** [comment]
retest this please

**Comment by shinobu-x** [comment]
retest this please

File: src/include/rados.h [hunk 1]
```diff
@@ -410,6 +410,8 @@ enum {
 						      pool uses pool snaps */
 	CEPH_OSD_FLAG_REDIRECTED   = 0x200000,  /* op has been redirected */
 	CEPH_OSD_FLAG_KNOWN_REDIR = 0x400000,  /* redirect bit is authoritative */
+	CEPH_OSD_FLAG_FULL_TRY =    0x800000,  /* try op despite full flag */
```
Comment (amitkumar50) [inline src/include/rados.h]: @shinobu-x (NIT) Please try to use title of PR < 50 characters for future PRs.
https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#title-of-pull-requests-and-title-of-commits

File: src/osdc/Objecter.cc [hunk 2]
```diff
@@ -898,7 +898,8 @@ bool Objecter::ms_dispatch(Message *m)
 
 void Objecter::_scan_requests(OSDSession *s,
                              bool force_resend,
-			     bool force_resend_writes,
+			     bool cluster_full,
+                             map<int64_t, bool> *pool_full_map,
```
Comment (amitkumar50) [inline src/osdc/Objecter.cc]: @shinobu-x Looks indent is incorrect

File: src/osdc/Objecter.cc [hunk 3]
```diff
@@ -2357,8 +2379,9 @@ bool Objecter::is_pg_changed(
 
 bool Objecter::target_should_be_paused(op_target_t *t)
 {
+  const pg_pool_t *pi = osdmap->get_pg_pool(t->base_oloc.pool);
```
Comment (amitkumar50) [inline src/osdc/Objecter.cc]: @shinobu-x (NIT) Coding guide says about const names:
https://google.github.io/styleguide/cppguide.html#Variable_Names
Your thought please?

**Comment by smithfarm** [comment]
Hammer is EOL.

**Pull request closed without merging**